### PR TITLE
Api update cols

### PR DESCRIPTION
**Pull request by mpage21**
Please look at this and let me know if you see anything that is missing or that should be changed. Or if you have questions about it

### CODE REVIEW

File: static/js/controllers/associations.controller.js [hunk 1]
```diff
@@ -6,14 +6,10 @@ var Request = require('request');
 exports.getFromTables = (req, res) => {
     var studyIDObjs = req.body.studyIDObjs
     var refGen = req.body.refGen;
-    var defaultSex = req.body.sex;
+    var sexes = req.body.sexes;
```
Comment (MattCloward) [inline static/js/controllers/associations.controller.js:9]: "sexes" should be "sex" for both the associations table and the study table

File: static/js/controllers/associations.controller.js [hunk 2]
```diff
@@ -6,14 +6,10 @@ var Request = require('request');
...
+    var ogValueType = req.ogValueType;
 
-    // if not given a defaultSex, default to exclude
-    if (defaultSex == undefined){
-        defaultSex = "e"
-    }
-
-    Association.getFromTables(studyIDObjs, refGen, async (err, data) => {
+    Association.getFromTables(studyIDObjs, refGen, sexes, ogValueType, async (err, data) => {
```
Comment (MattCloward) [inline static/js/controllers/associations.controller.js:12]: sexes -> sex

File: static/js/controllers/associations.controller.js [hunk 3]
```diff
@@ -24,20 +20,14 @@ exports.getFromTables = (req, res) => {
             associations = data[0]
             traits = data[1]
 
-            returnData = await separateStudies(associations, traits, refGen, defaultSex)
+            returnData = await separateStudies(associations, traits, refGen)
```
Comment (MattCloward) [inline static/js/controllers/associations.controller.js:23]: Is defaultSex here the sex requested by the user? If so, I believe we will still need it so the user can specify what studies they want to use.

File: static/js/controllers/associations.controller.js [hunk 4]
```diff
@@ -227,19 +219,21 @@ exports.joinTest = (req, res) => {
     })
 }
 
+//TODO update these last ones!
```
Comment (MattCloward) [inline static/js/controllers/associations.controller.js:222]: remove this TODO

File: static/js/controllers/associations.controller.js [hunk 5]
```diff
@@ -227,19 +219,21 @@ exports.joinTest = (req, res) => {
...
-    sex = req.query.defaultSex
+    sex = req.query.sex
 
     associationsPath = sex[0].toLowerCase() == "e" ? path.join(__dirname, '../..', `downloadables/associationsAndClumpsFiles/allAssociations_${refGen}.txt`) : path.join(__dirname, '../..', `downloadables/associationsAndClumpsFiles/allAssociations_${refGen}_${sex}.txt`)
     statsObj = fs.statSync(associationsPath)
     updateTime = statsObj.mtime
     res.send(`${updateTime.getFullYear()}-${updateTime.getMonth() + 1}-${updateTime.getDate()}`)
 }
 
+//ToDo should probably update this for pvalueAnnotations, and also get rid of sex?
```
Comment (MattCloward) [inline static/js/controllers/associations.controller.js:234]: remove the TODO?

File: static/js/controllers/associations.controller.js [hunk 6]
```diff
@@ -258,6 +252,7 @@ exports.getAssociationsDownloadFile = (req, res) => {
     }); 
 }
 
+//TODO should probably update this for pValueAnnotation!!!!!!!
```
Comment (MattCloward) [inline static/js/controllers/associations.controller.js:255]: remove the TODO?

File: static/js/controllers/studies.controller.js [hunk 7]
```diff
@@ -91,17 +91,41 @@ exports.getAll = (req, res) => {
...
         }
     });
 };
 
 exports.getFiltered = (req, res) => {
+    //TODO - need to also separate by pValueAnnotation
     traits = req.body.traits
     studyTypes = req.body.studyTypes
     ethnicities = req.body.ethnicities
+    sexes = req.body.sexes
```
Comment (MattCloward) [inline static/js/controllers/studies.controller.js:125]: sexes -> sex

File: static/js/controllers/studies.controller.js [hunk 8]
```diff
@@ -91,17 +91,41 @@ exports.getAll = (req, res) => {
...
+    ogValueTypes = req.body.ogValueTypes
     console.log("getting studies");
-    Study.getFiltered(traits, studyTypes, ethnicities, (err, data) => {
+    Study.getFiltered(traits, studyTypes, ethnicities, sexes, ogValueTypes, (err, data) => {
```
Comment (MattCloward) [inline static/js/controllers/studies.controller.js:128]: sexes -> sex

File: static/js/models/association.model.js [hunk 9]
```diff
@@ -8,20 +8,26 @@ const Association = function (massociation) {
...
+    this.betaAnnotation = massociation.betaAnnotation;
+    this.ogValueTypes = massociation.ogValueTypes;
     this.sex = massociation.sex;
+    this.numAssociationsFiltered = massociation.numAssociationsFiltered; //this should be going away and should never be used from this model
     this.citation = massociation.citation;
     this.studyID = massociation.studyID;
 };
 
-Association.getFromTables = (studyIDObjs, refGen, result) => {
+Association.getFromTables = (studyIDObjs, refGen, sexes, ogValueType, result) => {
```
Comment (MattCloward) [inline static/js/models/association.model.js:30]: sexes -> sex

File: static/js/controllers/studies.controller.js [hunk 10]
```diff
@@ -91,17 +91,41 @@ exports.getAll = (req, res) => {
...
+                    }
+                }
+            }
+            res.send(traitsList);
         }
     });
 };
 
 exports.getFiltered = (req, res) => {
+    //TODO - need to also separate by pValueAnnotation
```
Comment (MattCloward) [inline static/js/controllers/studies.controller.js:121]: I'll stop marking your TODO's, but be sure to check that the rest are resolved

File: static/js/controllers/associations.controller.js [hunk 11]
```diff
@@ -305,11 +302,20 @@ async function separateStudies(associations, traitData, refGen, defaultSex) {
...
+            studyIDsToMetaData[studyObj.studyID]['traits'][studyObj.trait] = {
+                studyTypes: traitStudyTypes,
+                pValBetaAnnoValType: [pvalBetaAnnoValType],
+                superPopulations: superPopulations
+            }
         }
         else {
-            studyIDsToMetaData[studyObj.studyID]['traits'][studyObj.trait] = traitStudyTypes
-            studyIDsToMetaData[studyObj.studyID]['ethnicity'] = Array.from(new Set([...studyIDsToMetaData[studyObj.studyID]['ethnicity'], ...ethnicities]))
+            // also change so that pValueAnno is an identifyier? and betaAnno??
```
Comment (MattCloward) [inline static/js/controllers/associations.controller.js:315]: identifyier -> identifier
Do you still need this comment? was it resolved?

File: static/js/controllers/associations.controller.js [hunk 12]
```diff
@@ -328,90 +334,31 @@ async function separateStudies(associations, traitData, refGen, defaultSex) {
...
-            }
-        }
-        // else add the pos->trait->studyID 
-        else if (association.studyID in studyIDsToMetaData){
-            // if there is no sex associated with the odds ratio or if the sex is the same as the requested sex (defaultSex)
-            if (association.sex == "NA" || association.sex[0].toLowerCase() == defaultSex[0].toLowerCase()) {
+        if (association.studyID in studyIDsToMetaData){
+            // if the pos/snp does not exist in our map and the studyID is in the associations
+            if (!(association.snp in AssociationsBySnp)){
+                AssociationsBySnp[association[refGen]] = association.snp // adds the pos to
```
Comment (MattCloward) [inline static/js/controllers/associations.controller.js:340]: incomplete comment

File: static/js/controllers/associations.controller.js [hunk 13]
```diff
@@ -328,90 +334,31 @@ async function separateStudies(associations, traitData, refGen, defaultSex) {
...
+            }
+            // if the pValBetaAnno not in the snp/trait/studyID obj
+            pValBetaAnnoValType = association.pValueAnnotation + "|" + association.betaAnnotation + "|" + association.ogValueTypes
+            if (!(pValBetaAnnoValType in AssociationsBySnp[association.snp]['traits'][association.trait][association.studyID])){
+                AssociationsBySnp[association.snp]['traits'][association.trait][association.studyID][pValBetaAnnoValType] = createStudyIDObj(association)
+            }
             else {
-                potentialDupliSnpsInfo.push([association.snp, association.trait, association.studyID])
+                console.log("Okay, we have a serious problem...")
+                // console.log(association, AssociationsBySnp[association.snp]['traits'][association.trait][association.studyID][pValBetaAnno])
```
Comment (MattCloward) [inline static/js/controllers/associations.controller.js:361]: remove commented out console.log

File: static/js/controllers/associations.controller.js (hunk 1 above)
Comment (mpage21) [reply static/js/controllers/associations.controller.js:9]: I'll make sure that is updated in the model. Here I think it is okay because the user could request to filter by Male and Female

File: static/js/controllers/associations.controller.js (hunk 3 above)
Comment (mpage21) [reply static/js/controllers/associations.controller.js:23]: They specify earlier. we don't need to separate the studies using that variable because that is already built in to the pValueAnnotation

File: static/js/controllers/associations.controller.js (hunk 11 above)
Comment (mpage21) [reply static/js/controllers/associations.controller.js:315]: yes

**Review by lizvance — APPROVED** [review]
Looks good to me!

File: static/js/controllers/associations.controller.js [hunk 14]
```diff
@@ -227,25 +219,24 @@ exports.joinTest = (req, res) => {
     })
 }
 
+//TODO DON'T FORGET TO UPDATE THE FILES THAT CREATE THESE!
```
Comment (MattCloward) [inline static/js/controllers/associations.controller.js:222]: remove this TODO

File: static/js/models/association.model.js [hunk 15]
```diff
@@ -8,20 +8,26 @@ const Association = function (massociation) {
...
     this.oddsRatio = massociation.oddsRatio;
     this.lowerCI = massociation.lowerCI;
     this.upperCI = massociation.upperCI;
-    this.pValueAnnotation = massociation.pValueAnnotation;
+    this.betaValue = massociation.betaValue;
+    this.betaUnit = massociation.betaUnit;
+    this.betaAnnotation = massociation.betaAnnotation;
+    this.ogValueTypes = massociation.ogValueTypes;
     this.sex = massociation.sex;
+    this.numAssociationsFiltered = massociation.numAssociationsFiltered; //this should be going away and should never be used from this model
```
Comment (MattCloward) [inline static/js/models/association.model.js:25]: Do you want me to remove this column from the tables?

File: static/js/models/database.js [hunk 16]
```diff
@@ -9,23 +9,25 @@ var dbConfig = {
...
+    pool = mysql.createPool(dbConfig);
 
-    console.log("We ran handleDisconnect...")
-    connection.connect(function(err) {
-        if (err) {
-            console.log('error when connecting to db: ', err);
-            setTimeout(handleDisconnect, 2000);
-        }
-    });
+    // console.log("We ran handleDisconnect...")
```
Comment (MattCloward) [inline static/js/models/database.js:22]: Can this commented code be removed?

File: static/js/models/study.model.js [hunk 17]
```diff
@@ -100,33 +107,42 @@ Study.getFiltered = (traits, studyTypes, ethnicities, result) => {
             result(null, null);
             return;
         }
-        var sqlQueryString = "";
-        sqlQueryParams = []
-        for (i = 0; i < res.length; i++) {
 
+        inner_callback = function(err) {
+            console.log("in inner callback")
+            // console.log(err)
```
Comment (MattCloward) [inline static/js/models/study.model.js:113]: can this comment be removed?

File: update_database_scripts/createServerAssociAndClumpsFiles.py [hunk 18]
```diff
@@ -11,6 +11,9 @@
 # How to run: python3 createServerAssociAndClumpsFiles.py "password"
 # where "password" is the password to the PRSKB database
 
+
+#TODO: This needs to be updated to reflect the new format. We should have one associations file per reference geneome
```
Comment (MattCloward) [inline update_database_scripts/createServerAssociAndClumpsFiles.py:15]: will this be done in another pull request?

**Review by MattCloward — CHANGES_REQUESTED** [review]
Just a few more comments for you and this looks good!

File: static/js/models/association.model.js (hunk 15 above)
Comment (mpage21) [reply static/js/models/association.model.js:25]: We can remove this later

File: static/js/models/database.js (hunk 16 above)
Comment (mpage21) [reply static/js/models/database.js:22]: Not yet, I'm still kind of testing this pool code and I want to keep this here just in case we have to go back to it
